### ISSUE TITLE
Adding Roles Via Class Expression Editor Is Throwing Syntax Exceptions/Errors

### ISSUE THREAD

**Issue by singhik** [NCIEVS/nci-protege5#492]
@bdionne  : Hi, 

Steps To Reproduce : 

1. Create any Concept f.e. 'TestR' 

![image](https://user-images.githubusercontent.com/36314371/101963835-becf4d00-3bdd-11eb-995b-4ea36d866d3c.png)

2. Add Roles via 'NCI Edit Tab'->'Description' 

![image](https://user-images.githubusercontent.com/36314371/101964005-09e96000-3bde-11eb-8da6-81c5ecb2552e.png)

'Class Expression Editor' is throwing an Exception on Syntax, though the Syntax is correct and Unable to add Roles to a Concept : 

![image](https://user-images.githubusercontent.com/36314371/101964082-2b4a4c00-3bde-11eb-8e50-c17727c19584.png)

Adding same via a Batch File works fine : 

![image](https://user-images.githubusercontent.com/36314371/101964278-d955f600-3bde-11eb-83ec-14cac9431275.png)

![image](https://user-images.githubusercontent.com/36314371/101964302-ef63b680-3bde-11eb-9e7c-05646647ac5e.png)

However, when the added Role is clicked, the Class Expression Editor Still throws Syntax Compilation Error/Exception ?? 

![image](https://user-images.githubusercontent.com/36314371/101964353-15895680-3bdf-11eb-9313-12224ba8650b.png)

**Comment by bdionne** [comment]
@singhik This is working as intented. When you select an existing parent to edit, it brings up a tabbed pane with only two panes. One lets you type in a parent (class expressin editor) and the other lets you find it in the taxonomy. In that context it's not allowing you to enter a role expression. If you click the add button in the subclass section you'll see a 4 tabbed pane and the class expression tab will allow role expressions.

In other words, it's context sensitive.

**Comment by singhik** [comment]
@bdionne  : Add button does **not** work with 4 tabbed pane , this issue is specific to 'Class expression Editor' throwing Syntax Exception when evaluating a valid Role Expression :: 

![image](https://user-images.githubusercontent.com/36314371/101970129-9bfd6280-3bf6-11eb-8471-267d6013ab50.png)

**Comment by bdionne** [comment]
I see, thanks @singhik - I'll look at it and see if it's a bug. That may also be the way it works. It is context dependent but I have used it to add roles before. Maybe the role and/or class is missing.

**Comment by bdionne** [comment]
@singhik ok, I was not able to reproduce this using my small test case, but **was** using the full Thesaurus. I'm not sure if or how we should fix this. It's actually working as intended! When you type in the name of the role, when it's parsing it, it checks first if the name is a class name. In your case it turns out that `Gene_Is_Element_In_Pathway` is both the name of a role (object property) and a class name. That's an artifact of the Thesaurus. Gilberto can explain why that is.

So what's happening is that it parses that word, interprets it as a class name and then expects to see a logical operator like `OR` or `AND` and not a role modifier like `SOME`.

This is coming from the Manchester parser down in the `OWL API` so I'm not sure if we should change this. I'll discuss with @NCIEVS, when we met tomorrow and let you know if there's anything we'll do.

Nice find!

**Comment by singhik** [comment]
As per discussion with @NCIEVS & @bdionne  , This is working as designed, the Exception can be done away by Rendering preferences with IRI and adding ROle and COncept codes in the Expression instead of labels or by  doing them with batch as Editors usually do .
